xds/src/main/java/com/linecorp/armeria/xds/BootstrapListeners.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.xds;
+
+import java.util.Map;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMap.Builder;
+
+import io.envoyproxy.envoy.config.bootstrap.v3.Bootstrap;
+import io.envoyproxy.envoy.config.bootstrap.v3.Bootstrap.StaticResources;
+import io.envoyproxy.envoy.config.listener.v3.Listener;
+
+final class BootstrapListeners {
+
+    private final Map<String, ListenerXdsResource> listeners;
+
+    BootstrapListeners(Bootstrap bootstrap) {
+        final Builder<String, ListenerXdsResource> builder = ImmutableMap.builder();
+        if (bootstrap.hasStaticResources()) {
+            final StaticResources staticResources = bootstrap.getStaticResources();
+            for (Listener listener: staticResources.getListenersList()) {
+                final ListenerXdsResource listenerResource = ListenerResourceParser.INSTANCE.parse(listener);
+                builder.put(listener.getName(), listenerResource);
```

Comment:
nit; envoy seems to generate a random name if not present.
I don't think this is an issue for now, but might be something to consider in case anyone files an issue.

ref: https://github.com/envoyproxy/envoy/blob/45b3f9af9d3889f23b4aded7416d92687cc72de0/source/common/listener_manager/listener_manager_impl.cc#L458